src/test/java/com/epam/task/purchases/base/BaseIOTest.java

```diff
@@ -0,0 +1,39 @@
+package com.epam.task.purchases.base;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public abstract class BaseIOTest {
```

Comment:
required functionality for testing in this task: 
`Euro.toString`, `getCost` in subclasses, `search` (you specify the argument right or not).